### ISSUE TITLE
The circlize package?

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#5968]
This could be added to the circular menu - and under the existing plotting dialogues in its own section.

Some examples with climatic data are provided in the MSc essay here.
[Legon_Seth_Owusu_aimsgh-essay-1516.pdf](https://github.com/africanmathsinitiative/R-Instat/files/5184769/Legon_Seth_Owusu_aimsgh-essay-1516.pdf)

Examples of the code for some line types are [here](https://www.r-graph-gallery.com/226-plot-types-for-circular-plot.html)

I am not sure, but assume that adding the circlize package is (just-about) worthwhile.  These are Circos Plots, which could be the name on the menu.  The circular part could be the months for daily data, or hours for hourly data (perhaps seasons, or 3-hrs if wanted fewer sectors round the circle.  Then there are Tracks.  These could either be Stations or Elements for a single Station.  Multiple plots are possible, so we could have Elements and then Facet by Station.

I suggest then a track is a Station then the type of graph will be the same for each track.  Hence I suggest the following structure be explored for the dialogue.

a) At the top there are 2 radio buttons with the label Tracks:  The first is Station and the second is Element.
b) Then there is our `usual data selector`, and under that is a button for `Circos Options`.  
c) The first (single) receiver has the label Sectors:
d) If Station is selected at the top the Second single selector is Station.  This is filled with the Station Field
e) Then there is `Plot Type`.  This is a pull Down.  Check on the different Options.  I think they are Bar, Box, Hist, Line, Point, Vline, Violin.
f) then Y Variable: - single receiver
g) then X Variable:  - single receiver
h) Then Facet:  Single receiver.

If Element is selected then we need the sort of control we have for the <right-click) Levels/Labels.  That replaces d), e) f) g) above.  It has 4 columns.  They are Element, Type, Y Variable, X Variable.

Then there is the Save Graph: at the bottom.
 We can start with defaults.  Then I am sure we will want some overall options on the main dialogue.  But there will also be many options for the sub-dialogue.  Perhaps there is a ,main Tab with label Overall.  Then there is a tab for each of the 7 plot types.

So, the main dialogue (at least) should be easy.  But it will really help if examples - with RStudio - are used to identify the R code and also what are sensible options for the sub-dialogue options.  And also to check that the structure of the dialogue holds up to most of the graphs that are sensible for the climatic data.

**Comment by IbrahimDanDije** [comment]
Well received dear Prof,
Best regards!
Ibrahim

**Comment by N-thony** [comment]
@rdstern and @IbrahimDanDije  I didn't find an empty dialog for circlize so I can design and implement at least there is under another name. For this issue #5968 I have to create a new dialog and design it first and enable Circlize under Structured>Circular>Circlize which is disabled.

**Comment by rdstern** [comment]
@N-thony yes, I meant to say there is already a space for it in the menu system, so you shouldn't need to work on frm_main.  It presumably needs a new dialogue
